src/error.rs

```diff
@@ -0,0 +1,31 @@
+use std::process::exit;
+
+/// Error message prefix
+const ERR: &str = "\x1b[31m\x1b[1merror\x1b[0m";
+
+/// Errors that might occur when attempting to interpret a program
+pub enum ErrorType {
+    ParseError,
+    IoError,
+    // CicrularDependency,
+    InternalError,
+}
+
+impl Default for ErrorType {
+    fn default() -> Self {
+        Self::InternalError
+    }
+}
+
+/// Function that is used to report a general compiler error
+pub fn report_error(err_type: ErrorType, err_msg: String) {
+    let prefix = match err_type {
+        ErrorType::ParseError => "Failed to parse\n",
+        // ErrorType::CicrularDependency => "Failed to import",
+        ErrorType::IoError => "Failed to read file",
+        ErrorType::InternalError => "Internal Panic", // @@TODO: add an internal panic function
+    };
+
+    println!("{}: {}{}", ERR, prefix, err_msg);
+    exit(-1);
```

Comment:
Hmm, maybe we shouldn't take the same approach we did with haskell---maybe we shouldn't immediately exit on error, but rather signal the error? Different parts of the compiler can their own error enums (with the help of `thiserror`) and return `Result`s. And then `main` can exit if it receives an error. This way we can do things like try perform some action and not necessarily exit. What do you think?

Reply:
Yes, I think this would be a better approach. I currently set it up to just 'mimic' the Haskell version. Of course, we should move away as much as possible from the Haskell version.